sbi/inference/snpe/snpe_base.py

```diff
@@ -263,10 +267,17 @@ def _train(
 
         Uses performance on a held-out validation set as a terminating condition (early
         stopping).
+
+        The proposal is only needed for non-atomic SNPE.
```

Comment:
Good catch with the `Optional[]`, I'm adding this.

As argued above, it's still `Any` though.

Setting default to `None` is not required because we always pass some value from `__call__()` (also all other arguments in `_train()` do not have defaults).